pkg/skaffold/deploy/helm/deploy.go

```diff
@@ -124,16 +124,20 @@ func (h *Deployer) Deploy(ctx context.Context, out io.Writer, builds []build.Art
 
 	// Deploy every release
 	for _, r := range h.Releases {
-		results, err := h.deployRelease(ctx, out, r, builds, valuesSet, h.bV)
+		releaseName, err := util.ExpandEnvTemplateOrFail(r.Name, nil)
+		if err != nil {
+			return nil, userErr(fmt.Sprintf("cannot expand release name %q", r.Name), err)
+		}
+		results, err := h.deployRelease(ctx, out, releaseName, r, builds, valuesSet, h.bV)
 		if err != nil {
-			releaseName, _ := util.ExpandEnvTemplate(r.Name, nil)
 			return nil, userErr(fmt.Sprintf("deploying %q", releaseName), err)
 		}
 
 		// collect namespaces
 		for _, r := range results {
 			var namespace string
-			namespace, err = util.ExpandEnvTemplate(r.Namespace, nil)
+			// `<no value>` is not allowed within a namespace
+			namespace, err = util.ExpandEnvTemplateOrFail(r.Namespace, nil)
```

Comment:
shd we  move this check before deploy for config  helm.releases[x].Namespace ?

Reply:
This namespace is [actually expanded and checked in `deployRelease()` itself](https://github.com/GoogleContainerTools/skaffold/blob/0a639aa25e65a4a5b448fdb21b10e91670d40f40/pkg/skaffold/deploy/helm/deploy.go#L355-L358).  So we're good on that front.

But this should be using [`h.releaseNamespace(r)`](https://github.com/GoogleContainerTools/skaffold/blob/0a639aa25e65a4a5b448fdb21b10e91670d40f40/pkg/skaffold/deploy/helm/util.go#L156) since the user could override the namespace on the command-line.

Reply:
Ah, we're ignoring the returned namespaces in our test.
https://github.com/GoogleContainerTools/skaffold/blob/0a639aa25e65a4a5b448fdb21b10e91670d40f40/pkg/skaffold/deploy/helm/helm_test.go#L966

Reply:
Hmm, this should be the deployed artifact's namespace, and so should be expanded.  Looking into this further.

Reply:
We should just be using the namespace returned from Helm.

This code path isn't tested, but let me see if I can add some tests for this.